docs/guides/benchmarks.md

```diff
@@ -0,0 +1,162 @@
+To mark such a function as a benchmark, we apply the `@benchmark` decorator.
+
+```python
+# benchmarks.py
+import nnbench
+from training import prepare_data, train_rf, accuracy
+
+@nnbench.benchmark()
+def benchmark_accuracy(n_estimators: int, max_depth: int, random_state: int) -> float:
+    X_train, X_test, y_train, y_test = prepare_data()
+    rf = train_rf(X_train=X_train, y_train=y_train, n_estimators=n_estimators,
+                  max_depth=max_depth, random_state=random_state)
+    acc = accuracy(model=rf, X_test=X_test, y_test=y_test)
+    return acc
+```
+
+Lastly, we set up a benchmark runner in the `run_benchmarks.py`. There, we supply the parameters (`n_estimators`, `max_depth`, `random_state`) necessary in the function definition. 
+
+```python
+# run_benchmarks.py
+from nnbench import runner
+
+r = runner.BenchmarkRunner()
+result = r.run("./benchmarks.py", params={"n_estimators": 100, "max_depth": 5, "random_state": 42})
+r.report(to='console', result=result)
+```
+
+When we execute the `run_benchmarks.py` we get the following output.
+
+
+```bash
+python run_benchmarks.py  
+
+
+name         value
+--------  --------
+accuracy  0.933333
+```
+
+# `@parametrize` for multiple configuration benchmarks
+
+Sometimes, we are not only interested in the performance of a model for given parameters but want to compare the performance for different configurations. 
+To achieve this, let's modify our `benchmarks.py` to run a parametrized benchmark. 
+We need to replace the decorator with our `@parametrize` configuration and supply the wanted parameter combinations within an iterable.
+
+```python
+# benchmarks.py
+import nnbench
+from training import prepare_data, train_rf, accuracy
+
+@nnbench.parametrize((
+    {"n_estimators": 10, "max_depth": 2},
+    {"n_estimators": 50, "max_depth": 5},
+    {"n_estimators": 100, "max_depth": 10}
+))
+def benchmark_accuracy(n_estimators: int, max_depth: int, random_state: int) -> float:
+    ... #   Our benchmarking logic from above
+```
+
+We still need to supply the remaining arguments to the `BenchmarkRunner()`, in our case the `random_state`.
+```python
+# run_benchmarks.py
+from nnbench import runner
+
+r = runner.BenchmarkRunner()
+result = r.run("./benchmarks.py", params={"random_state": 42})
+r.report(to='console', result=result)
+```
+
+Executing the parametrized benchmark, we get an output similar to this:
+
+```bash
+python run_benchmarks.py  
+
+
+name                                                 value
+------------------------------------------------  --------
+benchmark_accuracy_n_estimators=10_max_depth=2    0.955556
+benchmark_accuracy_n_estimators=50_max_depth=5    0.866667
+benchmark_accuracy_n_estimators=100_max_depth=10  0.911111
+```
+
+# `@product` for benchmark scans over parameter configurations
+
+In case we want to run a benchmark scan for all possible parameter combinations we can use the `@product` decorator and supply the different values for each parameter. 
```

Comment:
```suggestion
In case we want to run a benchmark scan for all possible combinations of a set of parameters, we can use the `@nnbench.product` decorator to supply the different values for each parameter. 
```